pkg/frontend/transport/handler.go

```diff
@@ -196,7 +196,7 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	if err != nil {
```

Comment:
I'm wondering why we wouldn't check the response status code here and create an error and return early. Also that would mean no change to repoertQueryStats - which is what I'd prefer. But I haven't worked on the query error handling recently. Possibly @aknuds1 could provide insight here?